src/Tools/Microsoft.dotnet-openapi/src/Internal/OpenapiDependencyAttribute.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.OpenApi;
+
+namespace Microsoft.DotNet.Openapi.Tools.Internal
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    internal class OpenApiDependencyAttribute : Attribute
+    {
+        public OpenApiDependencyAttribute(string name, string version, string codeGenerators)
+        {
+            Name = name;
+            Version = version;
+            CodeGenerators = codeGenerators.Split(';', StringSplitOptions.RemoveEmptyEntries).Select(c => Enum.Parse<CodeGenerator>(c));
```

Comment:
I'd do a `ToArray()` on this. Prevents callers from re-evaluating this multiple times over